### PR TITLE
Fix split particles

### PR DESCRIPTION
**Pull request by matthiasto**
We had an issue with particle splitting. The new particles were placed in the same place at the old one. Due to this he splited particles continued flowing as the old one did (as if it was one)...
This PR fixes this and introduces a new way of handling particles. Instead of splitting big particles (big in regard of mass) we split particles that have a too big area (either because of a big mass or because of a small flow depth). This ensures that we always have a stable number of particles within a kernel radius. This is all described in the hedgedoc and will be added to the doc.
This way of dealing with particles also questions the particle initialization. I think it should be modified and suggest an option in this PR.

https://hedgedoc.avaframe.org/XwbURAf3RRuldzUatWdhBA#

Also enable to initialize particles with a non constant flow depth

### CODE REVIEW

File: avaframe/com1DFA/particleTools.py
Comment (fso42) [inline avaframe/com1DFA/particleTools.py:341]: again: why not require it to be a numpy array?

File: avaframe/com1DFA/particleTools.py
Comment (fso42) [inline avaframe/com1DFA/particleTools.py:413]: needed?

File: avaframe/com1DFA/particleTools.py
Comment (fso42) [inline avaframe/com1DFA/particleTools.py:544]: test flag? or move to ini, or what is it needed for?

File: avaframe/com1DFA/particleTools.py
Comment (fso42) [inline avaframe/com1DFA/particleTools.py:1]: I really like the commenting and shortness/overview of the functions in this file!!

File: avaframe/in2Trans/shpConversion.py
Comment (fso42) [inline avaframe/in2Trans/shpConversion.py:319]: please add a test to this function

File: avaframe/out3Plot/outCom1DFA.py
Comment (fso42) [inline avaframe/out3Plot/outCom1DFA.py:187]: needed?

File: avaframe/out3Plot/outCom1DFA.py
Comment (fso42) [inline avaframe/out3Plot/outCom1DFA.py:178]: needed?

File: avaframe/out3Plot/outCom1DFA.py
Comment (fso42) [inline avaframe/out3Plot/outCom1DFA.py:191]: please add parameters

**Review by fso42 — CHANGES_REQUESTED** [review]
The function `initializeParticles` is hard to read. I suggest cleaning it up and moving quite some functionality to the `particleTools` (which is structured and commented nicely btw :+1: )

**Review by awirb — CHANGES_REQUESTED** [review]

File: avaframe/com1DFA/com1DFA.py
Comment (matthiasto) [reply avaframe/com1DFA/com1DFA.py:25]: Yes, plus DFA tools has not much to do with dfa anymore, it is partly functions to compute norms or scalar products and some functions to get normal and area on a dem raster. We coould rename it or split it even in two separate files. I am not a big fan of tools, it is not specific enough...

File: avaframe/com1DFA/particleTools.py
Comment (matthiasto) [reply avaframe/com1DFA/particleTools.py:174]: so if reasonString is '' it does not print anything. But we want the log.info if the reason is for example: because they were outside of the release area or because they exited the domain

File: avaframe/com1DFA/particleTools.py
Comment (matthiasto) [reply avaframe/com1DFA/particleTools.py:678]: this variable might be modified by this function, the track is then returned

File: avaframe/com1DFA/com1DFACfg.ini
Comment (matthiasto) [reply avaframe/com1DFA/com1DFACfg.ini:119]: The aim is to use the splitOption 0 with the MPPDH and the split option 2 with MPPK. The case you describe should then not really happen, only if somebody experiments

File: avaframe/com1DFA/com1DFA.py
Comment (fso42) [reply avaframe/com1DFA/com1DFA.py:25]: You want to tackle this in this PR right away? Or open an issue and wait until those other PR's are merged?

@name: I think tools is ok for exactly the reason of being a collection of unspecified functions. But if you have a more specific name in mind, please feel free to change it!

File: avaframe/com1DFA/particleTools.py
Comment (fso42) [reply avaframe/com1DFA/particleTools.py:176]: And what is the consequence for an end / expert user (i.e. non-developer; the target audience for the info-level)?
If it is so severe that some action is required, we should elevate it to Warning or even Error...

Background: RAMMS gives this info as well, and everybody just simply ignores it.

File: avaframe/com1DFA/com1DFA.py
Comment (fso42) [inline avaframe/com1DFA/com1DFA.py:125]: Since the adding of relThField  to add the functionality of having non uniform release heights is a big improvement, please add this as improvement in the PR description

File: avaframe/com1DFA/com1DFA.py
Comment (fso42) [inline avaframe/com1DFA/com1DFA.py:1312]: needed?

File: avaframe/com1DFA/com1DFA.py
Comment (fso42) [inline avaframe/com1DFA/com1DFA.py:1309]: ?? needed?

File: avaframe/com1DFA/com1DFATools.py
Comment (fso42) [inline avaframe/com1DFA/com1DFATools.py:17]: Please add a docstring

File: avaframe/com1DFA/testSPH.py
Comment (fso42) [inline avaframe/com1DFA/testSPH.py:217]: I think this is not wanted!! Self-reference :warning:

**Review by fso42 — CHANGES_REQUESTED** [review]
This is shaping up nicely. I think the functionality is good to go, some non-substantial comments are left. 

Only thing that is bigger: how do want to go on with the documentation for this? do you want add it here or open an issue an address it in another PR?

File: avaframe/com1DFA/com1DFA.py
Comment (matthiasto) [reply avaframe/com1DFA/com1DFA.py:1309]: This is an open question, I do not have the answer

Comment (fso42) [reply avaframe/com1DFA/com1DFA.py:1309]: ah, then at least remove the 'no big deal...' part

**Pull request merged**
